### ISSUE TITLE
for...in filter rule is easily broken

### ISSUE THREAD

**Issue by giladgray** [palantir/tslint#1695]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.0.6`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
for (let packageName in Examples) {
    const component = Examples[packageName][componentName];
    if (component != null) {
        return { component, packageName };
    }
}
```

with recommended `tslint.json` configuration.

#### Actual behavior

Fails to lint, saying `for...in statements must be filtered`.

#### Expected behavior

It is filtered, so I'd expect it to pass. Feels like this is a very common pattern (filtering by value instead of by key) so I'd expect the rule to detect the filter nonetheless.

**Comment by adidahiya** [comment]
It's unlikely that we'll update the logic of this archaic rule. I suggest you use `Object.keys(Examples)` instead with a for-of loop.

**Comment by Bidthedog** [comment]
Having a similar issue here:

```
        for (const groupName in formConfig) {
            if (!formConfig.hasOwnProperty(groupName)) {
                Debug.warn(`Cannot find group '${groupName}' in form config:\r\n${JSON.stringify(formConfig)}`);
                continue;
            }
       }
```

It passes fine when I remove the Debug.warn() statement - otherwise it fails with the 'for (... in ...) statements must be filtered with an if statement' error.

**Comment by JoshuaKGoldberg** [comment]
Per adidahiya's comment, closing this wontfix.
